src/tmt_web/utils/git_handler.py

```diff
@@ -73,14 +74,15 @@ def clone_repository(url: str, logger: Logger, ref: str | None = None) -> None:
 
 def get_path_to_repository(url: str) -> Path:
     """
-    Returns the path to the cloned repository from the given URL.
+    Returns the path to the cloned repository from the given URL. The repository url is hashed to
+    avoid repository name collisions.
 
     :param url: URL to the repository
     :return: Path to the cloned repository
     """
-    repo_name = url.rstrip("/").rsplit("/", 1)[-1]
+    clone_dir_name = hashlib.md5(url.encode()).hexdigest()
```

Comment:
I think we could still strip the `/` from the ending of the URL before the hashing.

Also, I am more for using `sha` (e.g., sha1/sha256) instead of `md5`, but it's not a strong opinion.

Reply:
Sure, I'll keep the strip. Imo `md5` is sufficient for this task, but no strong opinion either.

Reply:
Seems that [CodiumAI-Agent](https://github.com/CodiumAI-Agent) also does not like `md5` :grin:
So, if we already have two votes, let's use `sha` instead as it does not hurt?